Makefile

```diff
@@ -51,27 +52,36 @@ mbedtls_make:
 		if [ $$? != 0 ]; then exit 1; fi; \
 	done
 
+$(TALIB): $(UTILS_OBJS) $(CONNECTIVITY_OBJS) http_parser.o
+	$(CC) -shared -o $@ $<
+
 conn_http.o: connectivity/conn_http.c
 	@echo Compiling $@ ...
-	$(CC) -v -c $(CFLAGS) $(INCLUDES) -o $@ $<
-
+	$(CC) -c $(CFLAGS) $(INCLUDES) -o $@ $<
+http_parser.o: $(THIRD_PARTY_PATH)/http-parser/http_parser.c $(THIRD_PARTY_PATH)/http-parser/http_parser.h
+	$(CC) -c $(CFLAGS) $(INCLUDES) -o $@ $<
 main.o: main.c $(UTILS_OBJS)
 	@echo Compiling: $< ....
 	$(CC) -c $(CFLAGS) $(INCLUDES) -o $@ $<
 $(UTILS_OBJS):
 	$(MAKE) -C $(UTILS_PATH)
 test: $(TEST_PATH) $(UTILS_OBJS)
 	$(MAKE) -C $(TEST_PATH)
+ta_app: $(TALIB)
```

Comment:
May you change it to `ta_leagto_app` here, since I think we should state clearly that this rule would generate legato app only.